assets/javascripts/needleeditor.js

```diff
@@ -463,6 +500,17 @@ function saveNeedle(e) {
       addFlash('danger', '<strong>Unable to save needle:</strong><ul><li>' + errors.join('</li><li>') + '</li></ul>');
       return false;
   }
+
+  if ($("#property_workaround").prop("checked") && $("#input_workaround_desc").val() == "") {
+    var confirmMessage = "You set a workaround property for this needle, "
+                      + "but do not input description. "
+                      + "Are you sure you would not like to input any description for workaround?";
+    if (!confirm(confirmMessage)) {
```

Comment:
I'm ok with this but just want to note that we preferably use bootstrap dialogs. (You don't need to change it - I'm just saying.)

Reply:
Why shouldn't we be consistent with our use of bootstrap?

Reply:
I do not understand how to be consistent with our use of bootstrap, could you give me an example? I saw the other pages that need confirm message, also use "confirm" method.

Reply:
Yes, if it is not too important we just use the confirm method. Since it isn't even that ugly in modern browsers I also don't see a problem here except for the consistency with the few actual bootstrap dialogs we're using.

Reply:
An example in `needleeditor.js` would be `$('#modal-overwrite')`, another in `admin_needle.js` would be `$('#confirm_delete')` but I also agree that the current code is inconsistent in this and I'd leave it at `confirm` for now. It's not worth blocking the PR on.

Reply:
I would like to be not inconsistent with the bootstrap dialogs. In my opinion entering  any description about workaround or not is not mandatory, so one confirm is enough.